src/src/Helpers/HTTPResponseParser.cpp

```diff
@@ -0,0 +1,171 @@
+
+    if (str.length() > URI_MAX_LENGTH) {
+      addLog(LOG_LEVEL_ERROR, strformat(F("Response exceeds %d characters which could cause instabilities or crashes!"), URI_MAX_LENGTH));
+    }
+
+    auto processAndQueueParams = [](const String& url, const String& str, const String& eventName) {
+                                   // Extract the parameters from the URL
+                                   int start = url.indexOf(eventName + '=');
+
+                                   if (start == -1) {
+                                     return; // No parameters found for the given eventName
+                                   }
+                                   start += eventName.length() + 1;
+                                   const int end       = url.indexOf('&', start);
+                                   const String params = (end == -1) ? url.substring(start) : url.substring(start, end);
+
+                                   if (!params.isEmpty()) {
+                                     String keys[WEATHER_KEYS_MAX];
+                                     int    keyCount   = 0;
+                                     int    startIndex = 0;
+                                     int    commaIndex = params.indexOf(',');
+
+                                     // Split and add keys to the array
+                                     while (commaIndex != -1) {
+                                       if (keyCount >= WEATHER_KEYS_MAX) {
+                                         // Stop adding keys if array is full
+                                         addLog(LOG_LEVEL_ERROR,
+                                                strformat(F(
+                                                            "More than %d keys in the URL, this could cause instabilities or crashes! Try to split up the calls.."),
+                                                          WEATHER_KEYS_MAX));
+                                         break;
+                                       }
+                                       String key = params.substring(startIndex, commaIndex);
+                                       keys[keyCount++] = key;
+                                       startIndex       = commaIndex + 1;
+                                       commaIndex       = params.indexOf(',', startIndex);
+                                     }
+
+                                     // Add the last key
+                                     if (keyCount < WEATHER_KEYS_MAX) {
+                                       const String lastKey = params.substring(startIndex);
+                                       keys[keyCount++] = lastKey;
+                                     }
+
+                                     String csv;
+                                     const int startStringIndex = str.indexOf(strformat(F("\"%s\":"),
+                                                                                        eventName.c_str())) + eventName.length() + 4;
```

Comment:
With string operations like these, where you add a fixed offset, it is often a lot simpler to read when you add some example line as comment.